llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -102,6 +102,7 @@ SystemZTargetLowering::SystemZTargetLowering(const TargetMachine &TM,
     addRegisterClass(MVT::i32, &SystemZ::GR32BitRegClass);
   addRegisterClass(MVT::i64, &SystemZ::GR64BitRegClass);
   if (!useSoftFloat()) {
+    addRegisterClass(MVT::f16, &SystemZ::FP16BitRegClass);
```

Comment:
Do we want to add a VR16BitRegClass if we have vector support?

Reply:
So far I have a few points as "todo" for vector support:
  - add a VR16 reg class
  - use VLE/VSTE for reload/spill
  - Use generated vector constants.
 It wouldn't be too hard to add this, but not sure if it's relevant given all the slow conversion function calls. On the other hand, even if it's super-slow, maybe a noticeable speedup might have some value..?

Reply:
If it's not too hard, I think it might be good. Not so much because of performance, but also because it makes the code look more similar to what we do for the other types.